sklearn/model_selection/_search.py

```diff
@@ -453,8 +455,18 @@ def score(self, X, y=None):
             raise ValueError("No score function explicitly defined, "
                              "and the estimator doesn't provide one %s"
                              % self.best_estimator_)
-        score = self.scorer_[self.refit] if self.multimetric_ else self.scorer_
-        return score(self.best_estimator_, X, y)
+        if isinstance(self.scorer_, dict):
+            if self.multimetric_:
+                scorer = self.scorer_[self.refit]
+            else:
+                scorer = self.scorer_
```

Comment:
This is not covered. So it seems this when `scorer` is a dict but with a single metric?